metatensor-core/src/data.rs

```diff
@@ -93,6 +96,17 @@ pub struct mts_array_t {
         data: *mut *mut f64,
     ) -> mts_status_t>,
 
+    /// Get a DLPack representation of the underlying data.
+    ///
+    /// The returned `DLManagedTensorVersioned` is owned by the caller, who is
+    /// responsible for calling its `deleter` function when the tensor is no
+    /// longer needed. The lifetime of the `DLManagedTensorVersioned` must not
+    /// exceed the lifetime of the `mts_array_t` it was created from.
+    as_dlpack: Option<unsafe extern "C" fn(
+        array: *mut c_void,
+        dl_managed_tensor: *mut *mut DLManagedTensorVersioned,
```

Comment:
I think this function should take a `device` parameter, so we can automatically get CPU data for serialization. This is similar to the Python API: `__dlpack__` takes `dl_device`.

Similarly, we should think about the other arguments of `__dlpack__`:
- `stream` could make sense, not as something that we need in metatensor, but to give both sides of the code a way to communicate this information
- `max_version` makes sense to add IMO, for the same reasons it is in `__dlpack__`
- `copy` I'm not sure, this could be done by consumers after getting the dlpack array from us